Move ndb entities code snippets into github

jerjou opened the pull request.

jerjou commented:
/cc @jonparrott

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+
+def create_sandy_named_key():
+    account = Account(username='Sandy', userid=1234, email='some@where.com',
```
nit: please line break at the first `(`.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+def update_key(key):
+    sandy = key.get()
+    sandy.email = 'sandy@gmail.co.uk'
```
Please don't use real email addresses. Poor Sandy in the UK.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+
+def create_with_type_error_in_constructor():
+    bad = Account(username='Sandy',
```
nit: please line break at the first `(`

... in fact do this everywhere.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+
+def set_key_directly(account):
+    account.key = ndb.Key('Account', 'SOME@WHERE.COM')
```
If uppercase is not significant, I'd prefer you make the id lowercase.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+def create_sandy_named_key():
+    account = Account(username='Sandy', userid=1234, email='some@where.com',
+                      id='SOME@WHERE.COM')
```
please use `@example.com`, because that domain is specifically reserved for examples.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+def insert_message():
+    account_key = ndb.Key(Account, 'some@where.com')
+
+    # Lets ask Datastore to allocate an ID.
```
s/Lets a/A

It's a little *too* informal.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+def expando_properties(e):
+    return e._properties
+    # {'foo': GenericProperty('foo'), 'bar': GenericProperty('bar'),
```
Please feel free to format this better.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+
+def create_flex_employee():
+    emp = FlexEmployee(name='Sandy', location='SF')
```
please use more descriptive variable names.

jonparrott commented on appengine/ndb/entities/snippets.py:
```diff
@@ -0,0 +1,258 @@
+
+    @classmethod
+    def _post_delete_hook(cls, key, future):
+        notify('I suck and nobody likes me.')
```
uh... maybe something different here?

jonparrott commented on appengine/ndb/entities/snippets_test.py:
```diff
@@ -0,0 +1,194 @@
+    yield testbed
+
+    for name, obj in inspect.getmembers(snippets):
+        if inspect.isclass(obj) and issubclass(obj, ndb.Model):
```
`isclass` is redundant with issubclass.

jerjou replied:
Turns out it's not. issubclass raises an error if the first object is not a class.

jonparrott commented on appengine/ndb/entities/snippets_test.py:
```diff
@@ -0,0 +1,194 @@
+
+def test_create_sandy_with_keywords(client):
+    result = snippets.create_sandy_with_keywords()
+    assert type(result) == snippets.Account
```
Prefer `assert isinstance(result, snippets.Account)`

jonparrott commented:
I know these are snippets, but I'd like to see more docstrings/comments.

jerjou commented:
Addressed all comments.

I feel like docstrings/comments are kinda overkill for snippets. I renamed all the methods to be more descriptive, as a compromise. LMKWYT.

jonparrott commented:
LGTM
